tools/isobuild/bundler.js

```diff
@@ -1804,6 +1772,65 @@ class ClientTarget extends Target {
   }
 }
 
+const { wrap, defaultMakeCacheKey } = require("optimism");
+const minifyCssFiles = Profile("minifyCssFiles", wrap(function (files, {
+  arch,
+  minifier,
+  minifyMode,
+}) {
+  const sources = files.map(file => new CssFile(file, { arch }));
+  const markedMinifier = buildmessage.markBoundary(
+    minifier.userPlugin.processFilesForBundle,
+    minifier.userPlugin,
+  );
+
+  buildmessage.enterJob('minifying app stylesheet', function () {
+    try {
+      Promise.await(markedMinifier(sources, { minifyMode }));
+    } catch (e) {
+      buildmessage.exception(e);
+    }
+  });
+
+  return _.flatten(sources.map((source) => {
+    return source._minifiedFiles.map((file) => {
+      const newFile = new File({
+        info: 'minified css',
+        arch,
+        data: Buffer.from(file.data, 'utf8')
+      });
+      if (file.sourceMap) {
+        newFile.setSourceMap(file.sourceMap, '/');
+      }
+
+      if (file.path) {
+        newFile.setUrlFromRelPath(file.path);
+        newFile.targetPath = file.path;
+      } else {
+        newFile.setUrlToHash('.css', '?meteor_css_resource=true');
+      }
+
+      return newFile;
+    });
+  }));
+}, {
+  makeCacheKey(files, { arch, minifier, minifyMode }) {
+    return defaultMakeCacheKey(
+      minifier,
```

Comment:
The caching system uses the `minifier` object as a key in a `WeakMap` behind the scenes, so the cache will be invalidated and its memory will be reclaimed whenever a new minifier plugin is constructed. Fortunately, that doesn't happen on every rebuild, unless the implementation of the minifier has changed, requiring it to be rebuilt. In other words, rarely.